src/dscanner/analysis/imports_sortedness.d

```diff
@@ -5,89 +5,95 @@
 module dscanner.analysis.imports_sortedness;
 
 import dscanner.analysis.base;
-import dparse.lexer;
-import dparse.ast;
-
-import std.stdio;
 
 /**
  * Checks the sortedness of module imports
  */
-final class ImportSortednessCheck : BaseAnalyzer
+extern(C++) class ImportSortednessCheck(AST) : BaseAnalyzerDmd!AST
 {
 	enum string KEY = "dscanner.style.imports_sortedness";
 	enum string MESSAGE = "The imports are not sorted in alphabetical order";
 	mixin AnalyzerInfo!"imports_sortedness";
+	alias visit = BaseAnalyzerDmd!AST.visit;
 
 	///
-	this(string fileName, bool skipTests = false)
+	extern(D) this(string fileName)
 	{
-		super(fileName, null, skipTests);
+		super(fileName);
 	}
 
-	mixin ScopedVisit!Module;
-	mixin ScopedVisit!Statement;
-	mixin ScopedVisit!BlockStatement;
-	mixin ScopedVisit!StructBody;
-	mixin ScopedVisit!IfStatement;
-	mixin ScopedVisit!TemplateDeclaration;
-	mixin ScopedVisit!ConditionalDeclaration;
+	mixin ScopedVisit!(AST.StructDeclaration);
+	mixin ScopedVisit!(AST.FuncDeclaration);
+	mixin ScopedVisit!(AST.InterfaceDeclaration);
+	mixin ScopedVisit!(AST.UnionDeclaration);
+	mixin ScopedVisit!(AST.TemplateDeclaration);
+	mixin ScopedVisit!(AST.IfStatement);
+	mixin ScopedVisit!(AST.WhileStatement);
+	mixin ScopedVisit!(AST.ForStatement);
+	mixin ScopedVisit!(AST.ForeachStatement);
+	mixin ScopedVisit!(AST.ScopeStatement);
+	mixin ScopedVisit!(AST.ConditionalDeclaration);
+
 
-	override void visit(const VariableDeclaration id)
+	override void visit(AST.VarDeclaration vd)
 	{
 		imports[level] = [];
 	}
 
-	override void visit(const ImportDeclaration id)
+	override void visit(AST.Import i)
 	{
-		import std.algorithm.iteration : map;
+		import std.algorithm : map;
 		import std.array : join;
-		import std.string : strip;
+		import std.conv : to;
 
-		if (id.importBindings is null || id.importBindings.importBinds.length == 0)
-		{
-			foreach (singleImport; id.singleImports)
-			{
-				string importModuleName = singleImport.identifierChain.identifiers.map!`a.text`.join(".");
-				addImport(importModuleName, singleImport);
-			}
-		}
+		string importModuleName = i.packages.map!(a => a.toString().dup).join(".");
+
+		if (importModuleName != "")
+			importModuleName ~= "." ~ i.id.toString();
 		else
-		{
-			string importModuleName = id.importBindings.singleImport.identifierChain.identifiers.map!`a.text`.join(".");
+			importModuleName ~= i.id.toString();
 
-			foreach (importBind; id.importBindings.importBinds)
+		if (i.names.length)
+		{
+			foreach (name; i.names)
 			{
-				addImport(importModuleName ~ "-" ~ importBind.left.text, id.importBindings.singleImport);
+				string aux = to!string(importModuleName ~ "-" ~ name.toString());
+				addImport(aux, i);
 			}
 		}
+		else addImport(importModuleName, i);
 	}
 
-	alias visit = BaseAnalyzer.visit;
-
 private:
-
 	int level;
 	string[][int] imports;
+	bool[20] levelAvailable;
```

Comment:
Where is this magic number coming from?

Reply:
This is the maximum depth the check goes. Is it ok to have such number? For sure it would be better to extract it to a separate variable, but conceptually it is ok to have a maximum depth the check will go? I think it does not make sense to go further than 20 scopes in depth

Reply:
Yes, having a limit makes sense, but this should be defined in an enum.